src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -476,6 +476,9 @@ private struct ClosureInfo
             // Helper to decide whether we are inside the block or not
             public BlockInfo CurrentBlock;
 
+            // Dictionary for the used Labels in IL
+            public Dictionary<object, Label> Labels;

```

Comment:
I would prefer to use something more lightweight instead of dictionary. Given that we rarely will have 100+ labels.
I think an array of `KeyValuePair` with linear lookup will do.

Ideally I would like to go from arrays in `ClosureInfo` to something stack allocating, at least for small amount of items.